src/transport/SecureSessionMgr.cpp

```diff
@@ -431,7 +436,41 @@ void SecureSessionMgr::SecureMessageDispatch(const PacketHeader & packetHeader,
 
     if (packetHeader.GetDestinationNodeId().HasValue())
     {
-        admin->SetNodeId(packetHeader.GetDestinationNodeId().Value());
+        nodeId = packetHeader.GetDestinationNodeId().Value();
+        if (nodeId != kUndefinedNodeId && admin->GetNodeId() != nodeId)
+        {
+            admin->SetNodeId(nodeId);
+            ChipLogProgress(Inet, "Setting nodeID %" PRIX64 " on admin.", admin->GetNodeId());
+            modifiedAdmin = 1;
+        }
+        
+    }
+
+    if (packetHeader.GetSourceNodeId().HasValue())
+    {
+        fabricId = packetHeader.GetSourceNodeId().Value();
```

Comment:
No, that's not the fabric id... That's the source node id; not the same thing.

Reply:
once again see previous comments

Reply:
As above, document, link to tracking issue.

Reply:
left comments explaining